src/sipreg/reg.c

```diff
@@ -618,10 +611,13 @@ int sipreg_set_fbregint(struct sipreg *reg, uint32_t fbregint)
  */
 void sipreg_set_srcport(struct sipreg *reg, uint16_t srcport)
 {
-	if (!reg)
+	if (!reg || !reg->dlg)
 		return;
 
 	reg->srcport = srcport;
+
+	if (srcport)
```

Comment:
Is this check needed? calling `sipreg_set_srcport(reg, 0);` for srcport value reset should be fine?